### DISCUSSION TITLE
Does DB contain the VSLAM poses?

### DISCUSSION THREAD

**Discussion by mattiasmar** [introlab/rtabmap#1109] [Q&A]
Hello,

1) Is it correct to say that the rtabmap.db file doesn't contain the poses of the optimized graph, but rather only the raw measurements; odometry pose, detected links (loop closures), extracted feature points, the videos etc.?

2) If true, how can I control what optimization is done when calling the rtabmap-report tool? 
3) Is there a way to 'freeze' the optimization such that what I see during initial  rtabmap mapping, is the map that will remain with me in all further processing (i.e. reporting, localization)?

**Comment by matlabbe** [comment]
1. Yes and no, the optimized graph is saved on exit in [Admin](https://github.com/introlab/rtabmap/blob/6d36fc1fcd8e1f6855d245cbe8b104ba0b05516e/corelib/src/resources/DatabaseSchema.sql.in#L123-L124) table.It is used to avoid re-optimizing when re-loading the database. The corresponding occupancy grid map doesn't need to be regenerated too. That field is mainly used for ROS, and for iOS app.
2. All tools in rtabmap repo will always re-optimize the whole graph, following your observations in 1
3. If you want to save the final optimized graph when you created the database, you can open the database in rtabmap-databaseViewer, then select "Local Optimized" under graph view, then when doing File->Export poses, those poses will be exported.

![image](https://github.com/introlab/rtabmap/assets/2319645/96f44bcb-c1f7-455d-a683-6bfaec15909c)
